libfwupdplugin/fu-common.c

```diff
@@ -3378,3 +3384,102 @@ fu_xmlb_builder_insert_kb (XbBuilderNode *bn, const gchar *key, gboolean value)
 {
 	xb_builder_node_insert_text (bn, key, value ? "true" : "false", NULL);
 }
+
+/**
+ * fu_common_get_firmware_search_path:
+ * @error: (nullable): optional return location for an error
+ *
+ * Reads the FU_PATH_KIND_FIRMWARE_SEARCH and
+ * returns its contents
+ *
+ * Returns: a pointer to a gchar array
+ *
+ * Since: 1.6.2
+ **/
+gchar *
+fu_common_get_firmware_search_path (GError **error)
+{
+	gsize sz = 0;
+	g_autofree gchar *sys_fw_search_path = NULL;
+	g_autofree gchar *contents = NULL;
+
+	sys_fw_search_path = fu_common_get_path (FU_PATH_KIND_FIRMWARE_SEARCH);
+	g_file_get_contents(sys_fw_search_path, &contents, &sz, error);
+
+	/* remove newline character */
+	if (contents != NULL && sz > 0 && contents[sz - 1] == '\n')
+		contents[sz - 1] = 0;
+
+	g_debug ("read firmware search path: %s", contents);
+
+	return g_steal_pointer (&contents);
+}
+
+/**
+ * fu_common_set_firmware_search_path:
+ * @path: string to write
+ * @error: (nullable): optional return location for an error
+ *
+ * Writes to the FU_PATH_KIND_FIRMWARE_SEARCH
+ *
+ * Returns: %TRUE if successful
+ *
+ * Since: 1.6.2
+ **/
+gboolean
+fu_common_set_firmware_search_path (const gchar *path, GError **error)
+{
+#if GLIB_CHECK_VERSION(2,66,0)
+	g_autofree gchar *sys_fw_search_path_prm = NULL;
+	gsize len = strlen (path);
+
+	g_return_val_if_fail (path != NULL, FALSE);
+
+	if (len == 0)
+		len = 1;
```

Comment:
eigh?

Reply:
If we pass an empty string to the function, meaning that we want to erase the contents of the firmware search path, strlen will return 0 and `g_file_set_contents_full` will not change the contents of the file. Hence len is changed to 1 so 0 is written to the file.
I couldn't think of any other way to overcome this

Reply:
i guess i can add the len parameter to the function arguments and do this checking in the calling function

Reply:
or maybe add another function to clear the firmware search path. please suggest what would be a better solution here

Reply:
Yes, I think another function to unset would be much better -- e.g. fu_common_reset_firmware_search_path which also means we can use it from a FuDeviceLocker and do the right thing on the error path in all cases.

Reply:
So I added `fu_common_reset_firmware_search_path` to FuCommon and used FuDeviceLocker for setting up and restoring firmware path during firehose update